### PR TITLE
Adjusts the way load works

### LINKED ISSUE #296: Save Load Maintenance
### **Feature:** #125

Saving and loading needs a lot of maintenance as when new features they need to be added or ignored to the save load methods and as such this feature ticket will be used to keep track of what was been saved and what is to be saved. The list below in milestones will not include those that were completed as part of the original feature ticket #125. Will be adding the label for any team that makes work that needs to be saved in order to better collaborate.
 
# **Milestones**
List of steps that need to be completed for this task. This will be expanded as more things that need to be saved are made.
- [x] Step 1 (TBD) - Fix saving inventory (waiting for inventory rework)
- [x] Step 2 (Sep. 26) - Add saving quests (waiting for additional features)
- [x] Step 3 (Sep. 26) - Fix saving weather events (Not correctly saving it at the moment)
- [x] Step 4 (Sep. 14) - Fixes the plant naming convention to match the ones in FactoryService.
- [ ] Step 5 (TBD) - Implement saving health (waiting for health to be made)
- [x] Step 6 (Sep. 21) - Saves minutes from time service
- [x] Step 7 (Sep. 26) - Redo save/load to use cascading read calls
- [x] Step 8 (Oct 3) - Saves/loads placeables
 
**Completion Deadline:** N/A
 
# **Documentation**
- https://github.com/UQcsse3200/2023-studio-1/wiki/Save---Load-game
 
# **Member**
Jackson Trenerry (@JTrenerry)

### PR DESCRIPTION
**Pull request by JTrenerry**
A new method is used to load Entities and other things in the game, should not break any existing code but please feel free to test it on the branch before approving. This allowed the removal of many getters and setters and as such making the private variables of a class actually feel private. Fixes #296

### CODE REVIEW

**Comment by JTrenerry** [comment]
Found a better solution for ClimateController and will be updated soon

**Comment by JTrenerry** [comment]
The change to ClimateController could technically be done to TimeService but TimeService has semi-important data within it and should be cleared on load rather than just updating previous and retaining that info.

File: source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java [hunk 1]
```diff
@@ -224,19 +224,11 @@ public void create() {
     spawnTerrain();
     spawnInvisibleObstacle();// spawn invisible obstacle on the non-traversable area of the map
 
-    spawnCrop(5, 11, "Cosmic Cob");
-    spawnCrop(7, 11, "Aloe Vera");
-    spawnCrop(9, 11, "Hammer Plant");
-    spawnCrop(11, 11, "Space Snapper");
-    spawnCrop(13, 11, "Deadly Nightshade");
-    spawnCrop(15, 11, "Atomic Algae");
```
Comment (86LAK) [inline source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java:232]: hmm

File: source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java [hunk 2]
```diff
@@ -315,10 +307,6 @@ private void spawnInvisibleObstacle() {
      }
    }
 
-   private void spawnFirefly() {
```
Comment (86LAK) [inline source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java:318]: Firefly go brrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr

**Review by 86LAK — APPROVED** [review]
Didnt test it but the code  LGTM

File: source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java (hunk 1 above)
Comment (JTrenerry) [reply source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java:232]: Was incorrectly spawned so removed them

File: source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java (hunk 2 above)
Comment (JTrenerry) [reply source/core/src/main/com/csse3200/game/areas/SpaceGameArea.java:318]: Was leftover test code

**Review by lucashicks1 — APPROVED** [review]
LGTM. With toggling tractor lights, does the tractor have both light types and that's why you are not using events for light toggle? Just curious.

**Comment by JTrenerry** [comment]
> LGTM. With toggling tractor lights, does the tractor have both light types and that's why you are not using events for light toggle? Just curious.

Exactly! There is an AuraLightComponent whenever you in the tractor (lights from like inside the tractor ig) and the headlights (ConeLightComponent) which I want to be toggled with a key and stay on outside the tractor, so events did not work.

**Comment by lucashicks1** [comment]
> > LGTM. With toggling tractor lights, does the tractor have both light types and that's why you are not using events for light toggle? Just curious.
> 
> 
> 
> Exactly! There is an AuraLightComponent whenever you in the tractor (lights from like inside the tractor ig, the ones your dad would yell at you for having on) and the headlights (ConeLightComponent) which I want to be toggled with a key and stay on outside the tractor, so events did not work.

Oh okay that's what I was thinking. I might add other light specific events (eg: "toggleAuraLight" and "toggleConeLight") so that events can still work in this case, and just so it's cleaner. I didn't think about entities with two lights whoops. When I've got my laptop on me I'll make the changes and tag you in a PR.

**Review by DonghyugDavidJeong — APPROVED** [review]

**Review by p-lahanas — APPROVED** [review]
lgtm

**Review by cmp6 — APPROVED** [review]
nice 👍

**Pull request merged**